### DISCUSSION TITLE
Sample code for lab.js

### DISCUSSION THREAD

**Discussion by kurokida** [kurokida/jsQuestPlus#2] [Ideas]
Hi @tpronk , I made a sample code for lab.js.
It's working fine, but I have two concerns.

I have attached a sample code, and I would like to ask you to proofread it in English.
First, please convert the extension of [study-2021-12-07--15_41.study.txt](https://github.com/kurokida/jsQuestPlus/files/7666140/study-2021-12-07--15_41.study.txt) from txt to json.
Second, launch [the lab.js](https://lab.js.org/) builder and open the json file.

I was also able to export [the file for pavlovia](https://github.com/kurokida/jsQuestPlus/files/7666146/study-pavlovia.zip). 
This file may be easier for you to use.
There is no difference in the content of the program between the json file and the pavlovia files.

The other concern is about CDN servers.
The sample code for lab.js refers to jsQuestPlus.js on my university's server.
In general, it is better to refer to the file on the CDN server, but I am not familiar with it.
Do you have any idea?

Best, Daiichiro

**Comment by tpronk** [comment]
Hey @kurokida,

I copy-pasted the text into word and made some changes with "track changes" enabled. Would this help you out?
[jsQuestPlus Demo Texts.docx](https://github.com/kurokida/jsQuestPlus/files/7684894/jsQuestPlus.Demo.Texts.docx)

About CDNs. I'm not experienced with them, but what I read [here](https://stackoverflow.com/questions/44981404/how-do-i-upload-to-cdnjs) makes me suspect we're not big enough yet. In PsychoJS, you can upload the library to an experiment repo, which solves the problem. Perhaps lab.js has some feature for bundling JS libraries with an experiment? Shall we ask in their Slack channel?

Best, Thomas

**Reply by kurokida** [reply]
Thank you for your prompt response.

I have changed the English text. It was very informative.

I understood we don't need to upload the file to the CDN server. I continue to open the file on my server as it is now.

I'll ask about bundling JS files in a pull request to lab.js.

Best, Daiichiro
